cpp/CMakeLists.txt

```diff
@@ -339,16 +355,31 @@ message(STATUS "CUDF_TEST_LIST set to: ${CUDF_TEST_LIST}")
 # - add google benchmark --------------------------------------------------------------------------
 
 if(BUILD_BENCHMARKS)
+    
+    if(benchmark_ROOT)
+        message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${benchmark_ROOT}")
+        add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
+    else()
+        message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found, it will be built from source.")
+        include(ConfigureGoogleBenchmark)
+        add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
+        set(benchmark_ROOT "${CMAKE_BINARY_DIR}/googlebenchmark")
+    endif(benchmark_ROOT)
 
-  include(ConfigureGoogleBenchmark)
+    find_package(benchmark)
```

Comment:
This should be moved above the `if(benchmark_ROOT)` similar to what you have for `gtest`